apps/sms/views/shared/js/navigation.js

```diff
@@ -573,36 +573,43 @@ function switchPanel() {
   var isGoingBack = oldView && oldView.previous === newView.name;
 
   newPanelElement.style = '';
+  newPanelElement.classList.remove('panel-hidden');
+  newPanelElement.classList.add('panel-active');
 
   var doSlideAnimation =
     from && from.panel &&
     from.panel.behavior !== newView.behavior;
 
+  var doFadeInAnimation = !from || !from.panel;
+
   var animationPromise;
   if (doSlideAnimation) {
-
     var shouldGoRight = isGoingBack ^ (document.dir === 'rtl');
     newPanelElement.style.animationName =
-      shouldGoRight ? 'new-slide-right' : 'new-slide-left';
+      shouldGoRight ? 'left-to-current' : 'right-to-current';
     oldPanelElement.style.animationName =
-      shouldGoRight ? 'old-slide-right' : 'old-slide-left';
+      shouldGoRight ? 'current-to-right' : 'current-to-left';
+  } else if (doFadeInAnimation) {
+    newPanelElement.style.animationName = 'fade-in';
+  }
 
+  if (doFadeInAnimation || doSlideAnimation) {
     animationPromise = waitForSlideAnimation(newPanelElement).catch(
       () => {}
     ).then(() => {
-      oldPanelElement.style = '';
+      if (oldPanelElement) {
+        oldPanelElement.style = '';
+      }
       newPanelElement.style = '';
     });
   }
 
   return (animationPromise || Promise.resolve()).then(() => {
-    if (oldView) {
+    if (oldPanelElement && oldPanelElement !== newPanelElement) {
```

Comment:
mmm, should not we just do just something like the snippet below before any manipulations with classes and aria attributes (sorry if I'm missing something obvious _again_)?
```
if (newPanelElement === oldPanelElement) {
  return Promise.resolve();
}
```